client/components/select-vertical/suggestion-search.tsx

```diff
@@ -21,35 +23,71 @@ const SelectVerticalSuggestionSearch: FC< Props > = ( {
 	placeholder,
 	searchTerm,
 	suggestions,
+	featuredSuggestions,
 	isDisableInput,
 	isLoading,
 	onInputChange,
 	onSelect,
 } ) => {
 	const [ isShowSuggestions, setIsShowSuggestions ] = useState( false );
 	const [ isFocused, setIsFocused ] = useState( false );
+	const inputRef = useRef( null );
 	const suggestionsRef = useRef( null );
+	const toggleIconRef = useRef( null );
 	const translate = useTranslate();
 
-	const handleTextInputBlur = useCallback( () => {
-		setIsFocused( false );
-		setIsShowSuggestions( false );
-	}, [ setIsShowSuggestions, setIsFocused ] );
+	const handleTextInputBlur = useCallback(
+		( event ) => {
+			if ( event && event.relatedTarget?.contains( toggleIconRef.current ) ) {
+				return;
+			}
 
-	const handleTextInputFocus = useCallback( () => {
-		if ( suggestions.length > 0 ) {
-			setIsShowSuggestions( true );
-		}
+			setIsShowSuggestions( false );
+			setIsFocused( false );
+		},
+		[ setIsShowSuggestions, setIsFocused ]
+	);
 
+	const handleTextInputFocus = useCallback( () => {
+		setIsShowSuggestions( true );
 		setIsFocused( true );
-	}, [ suggestions, setIsFocused, setIsShowSuggestions ] );
+	}, [ setIsShowSuggestions, setIsFocused ] );
 
 	const handleTextInputChange = useCallback(
-		( event: ChangeEvent< HTMLInputElement > ) => {
-			setIsShowSuggestions( event.target.value.trim().length > 0 );
+		( event: React.ChangeEvent< HTMLInputElement > ) => {
+			setIsShowSuggestions( true );
 			onInputChange?.( event.target.value );
 		},
-		[ setIsShowSuggestions, onInputChange ]
+		[ setIsShowSuggestions ]
+	);
+
+	const handleToggleSuggestionsBlur = useCallback(
+		( event ) => {
+			if ( event && event.relatedTarget?.contains( inputRef.current ) ) {
```

Comment:
What does this check want to do?

Reply:
This check is to keep the suggestion dropdown open when the focus moves between the input field and the toggle icon (button). I've added comments to explain the intention.